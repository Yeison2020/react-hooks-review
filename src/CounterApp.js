import PropTypes from "prop-types";
import { useState } from "react";
// Note I needed to add this default value to make my test pass Learn why ?

function CounterApp({ value = 10 }) {
  const [counter, setCounter] = useState(value);

  // Handle Functions
  const handleAdd = () => {
    // When you dont no know the name of the counter
    // No need to return;
    setCounter((c) => c + 1);
  };

  const handleReset = () => {
    setCounter(0);
  };

  const handleReduce = () => {
    setCounter(counter - 1);
  };

  //-------------Returns
  return (
    <div>
      <h1>Counter App</h1>
      <h2>{counter}</h2>
      <button onClick={handleAdd}>+1</button>
      <button onClick={handleReset}>Reset</button>
      <button onClick={handleReduce}>-1</button>
    </div>
  );
}

CounterApp.propTypes = {
  value: PropTypes.number.isRequired,
};

export default CounterApp;
