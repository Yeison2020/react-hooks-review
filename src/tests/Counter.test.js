import React from "react";
import { shallow } from "enzyme";
import CounterApp from "../CounterApp";
import "@testing-library/jest-dom";

//----------Counter App testing

describe("Pruebas en Counter Component", () => {
  let wrapper = shallow(<CounterApp />);
  beforeEach(() => {
    wrapper = shallow(<CounterApp />);
  });

  //first test-----------------------------

  test("It should show counter componet matching with the Snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  //-------------------------------------

  test("Tiene que mostrar el value default of 10", () => {
    const wrapper = shallow(<CounterApp value={10} />);
    const counterText = wrapper.find("h2").text().trim();
    expect(counterText).toBe("10");
    //Here is why I got the Error The text returned string
  });

  test("Should increment the counter", () => {
    wrapper.find("button").at(0).simulate("click");
    const textCounter = wrapper.find("h2").text().trim();
    expect(textCounter).toBe("11");
  });

  test("This should decrement the counter by 1", () => {
    wrapper.find("button").at(2).simulate("click");
    const counterBtn2 = wrapper.find("h2").text().trim();
    expect(counterBtn2).toBe("9");
  });

  test("This button should Reset the value of the counter", () => {
    wrapper.find("button").at(0).simulate("click");
    const textCounter = wrapper.find("h2").text().trim();
    console.log(textCounter);
    //-------------------
    wrapper.find("button").at(1).simulate("click");
    const btnReset = wrapper.find("h2").text().trim();
    expect(btnReset).toBe("0");
  });
});
