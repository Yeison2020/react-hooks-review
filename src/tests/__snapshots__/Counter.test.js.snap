// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en Counter Component It should show counter componet matching with the Snapshot 1`] = `
<div>
  <h1>
    Counter App
  </h1>
  <h2>
    10
  </h2>
  <button
    onClick={[Function]}
  >
    +1
  </button>
  <button
    onClick={[Function]}
  >
    Reset
  </button>
  <button
    onClick={[Function]}
  >
    -1
  </button>
</div>
`;
